File: Grind 75 in Java/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hm = new HashMap<Integer, Integer>();
        int[] res = new int[2];

        for(int i = 0; i < nums.length; i++) {
            if(hm.containsKey(target - nums[i])) {
                res[0] = i;
                res[1] = hm.get(target - nums[i]);
                break;
            } else {
                hm.put(nums[i], i);
            }
        }

        return res;
    }
}

File: leetcode in kotlin/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {      
        val res = intArrayOf(-1, -1)
        val map = HashMap<Int, Int>()
        
        for (i in nums.indices) {
            if (map.containsKey(nums[i])) {
                res[0] = i
                res[1] = map[nums[i]]!!
                break
            } else {
                map[target - nums[i]] = i
            }
        }

        return res
    }
}
